extras/devices/base.py

```diff
@@ -0,0 +1,35 @@
+import logging
+
+logger = logging
+
+
+class AbstractDevice:
```

Comment:
For abstract classes, you can use ABC base class. See https://docs.python.org/3/library/abc.html

Reply:
Done. Thanks for the reference :)